scripts/js/upgrade/index.js

```diff
@@ -0,0 +1,330 @@
+    await waitUntilEventFound(api, "ExtrinsicSuccess")
+
+    // Wait one extra session
+    await waitUntilEventFound(api, "EmptyTerm")
+    // await waitUntilEventFound(api, "NewSession")
+
+    const keyring = new Keyring({ type: "sr25519" });
+    const alice = keyring.addFromUri(seeds[0]);
+    const bob = keyring.addFromUri(seeds[1]);
+    const charlie = keyring.addFromUri(seeds[2]);
+
+    const wasm = fs.readFileSync(wasmFile)
+    const wasmHash = blake2AsHex(wasm)
+    console.log("Applying WASM Blake2 Hash:", wasmHash)
+
+    let nonce = Number((await api.query.system.account(alice.address)).nonce);
+    const preimageNoted = await notePreimageAuth(api, alice, wasmHash, nonce);
+
+    console.log("Continuing with council proposal using", preimageNoted)
+    nonce = Number((await api.query.system.account(alice.address)).nonce);
+    const result = await councilProposeDemocracy(api, alice, preimageNoted, nonce)
+
+    console.log("Continuing with council vote using", result[0], result[1])
+    nonce = Number((await api.query.system.account(alice.address)).nonce);
+    await councilVoteProposal(api, alice, result[0], result[1], nonce, false)
+    nonce = Number((await api.query.system.account(bob.address)).nonce);
+    await councilVoteProposal(api, bob, result[0], result[1], nonce, false)
+    nonce = Number((await api.query.system.account(charlie.address)).nonce);
+    await councilVoteProposal(api, charlie, result[0], result[1], nonce, true)
+
+    console.log("Continuing to close council vote")
+    nonce = Number((await api.query.system.account(alice.address)).nonce);
+    const democracyIndex = await councilCloseProposal(api, alice, result[0], result[1], nonce)
+
+    console.log("Continuing with democracy vote on ref index", democracyIndex)
+    nonce = Number((await api.query.system.account(alice.address)).nonce);
+    await voteReferenda(api, alice, democracyIndex, nonce)
+
+    console.log("Waiting for referenda to be over and UpgradeAuthorized event is triggered")
+    await waitUntilEventFound(api, "UpgradeAuthorized")
+
+    console.log("Proceeding to enact upgrade")
+    nonce = Number((await api.query.system.account(alice.address)).nonce);
+    await enactUpgrade(api, alice, `0x${wasm.toString('hex')}`, nonce);
+
+    console.log("Waiting for ValidationFunctionApplied event")
+    await waitUntilEventFound(api, "ValidationFunctionApplied")
+
+    console.log("Waiting for 3 NewSession events")
+    let foundInBlock = await waitUntilEventFound(api, "EmptyTerm")
+    // await waitUntilEventFound(api, "NewSession")
+
+    console.log("getting header")
+    console.log("First event found, waiting for second event")
+    foundInBlock = await waitUntilEventFound(api, "EmptyTerm", foundInBlock+1)
+    // await waitUntilEventFound(api, "NewSession")
+
+    console.log("Second event found, waiting for third event")
+    await waitUntilEventFound(api, "EmptyTerm", foundInBlock+1)
+    // await waitUntilEventFound(api, "NewSession")
```

Comment:
NIT: You could run this on a while block instead of duplicating the code 3 times.